### PR TITLE
Add S3 inputs to "artifacts" directory

### PR DESCRIPTION
**Pull request by prakashsurya**
This change helps us preserve the inputs to the build that come from S3.
Previously, after the files from S3 were downloaded and processed, they
would be deleted. Now, we preserve the files within the "artifacts"
file, such that they can be preserved by any automation running the
build, just like we've been doing with the outputs of the build.

### CODE REVIEW

**Comment by prakashsurya** [comment]
`git-ab-pre-push` is [here](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/2988/)

File: scripts/build-ancillary-repository.sh [hunk 1]
```diff
@@ -75,21 +75,34 @@ function resolve_s3_uri() {
...
+	# artifacts directory, such that these build inputs can be
+	# preserved along with all of the build outputs. This way, we
+	# have a better way of later auditing the build outputs, by
+	# comparing it with the build inputs. Additionally, this helps
+	# any effort to regenerate the build outputs from the same exact
+	# build inputs, as we could later re-run the build using the
+	# same exact S3 artifacts used during the original build.
+	#
+	mkdir -p "$TOP/build/artifacts/inputs"
+	mv "$tmp_directory" "$TOP/build/artifacts/inputs/$input_name"
```
Comment (pzakha) [inline scripts/build-ancillary-repository.sh]: As a bonus, could we also include a text file to each input's directory (`$TOP/build/artifacts/inputs/$input_name/`) with the S3 URL used to fetch the input?

Comment (prakashsurya) [reply scripts/build-ancillary-repository.sh]: Yea, sure; that seems reasonable.

**Comment by prakashsurya** [comment]
Closing in favor of a solution where our Jenkins automation will handle this; see [here](http://reviews.delphix.com/r/56351).

**Pull request closed without merging**